app/models/user.rb

```diff
@@ -1,6 +1,15 @@
 # frozen_string_literal: true
+require 'friendship_methods'
 
 class User < ApplicationRecord
+  include FriendshipMethods
+
+  has_many :friendships
+  has_many :friends, through: :friendships
+  has_and_belongs_to_many :friends
```

Comment:
You don't need this type of association here because you are already using one approach with the `:through` association.

Reply:
removed